app/src/main/java/cm/aptoide/pt/view/DeepLinkManager.java

```diff
@@ -263,10 +283,23 @@ private void appViewDeepLink(long appId, String packageName, boolean showPopup,
     appNavigator.navigateWithAppId(appId, packageName, openType, "", oemId);
   }
 
-  private void appViewDeepLink(String packageName, String storeName, boolean showPopup) {
-    AppViewFragment.OpenType openType = showPopup ? AppViewFragment.OpenType.OPEN_WITH_INSTALL_POPUP
-        : AppViewFragment.OpenType.OPEN_ONLY;
-    appNavigator.navigateWithPackageAndStoreNames(packageName, storeName, openType);
+  private void appViewDeepLink(String packageName, String storeName, String openType) {
+    appNavigator.navigateWithPackageAndStoreNames(packageName, storeName, getOpenType(openType));
+  }
+
+  private AppViewFragment.OpenType getOpenType(String openType) {
+    if (openType == null) return AppViewFragment.OpenType.OPEN_ONLY;
+    switch (openType) {
+      case "open_and_install":
```

Comment:
shouldnt these be a field ? So that it is less error prone while trying to set it to use it?